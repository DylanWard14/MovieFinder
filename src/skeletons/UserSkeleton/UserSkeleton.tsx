import React from 'react';
import styled, { keyframes } from 'styled-components/macro';

const Component = styled.div`
    display: flex;
    align-items: center;
`;

const load = keyframes`
from {
    left: -150px;
}
to   {
    left: 100%;
}
`;

const Avatar = styled.div`
    width: 2rem;
    height: 2rem;
    background: #465c52;
    margin-left: 0.5rem;
    border-radius: 50%;
    position: relative;
    overflow: hidden;
    &::before {
        content: '';
        display: block;
        position: absolute;
        left: -150px;
        top: 0;
        height: 100%;
        width: 150px;
        background: linear-gradient(to right, transparent 30%, #009952 50%, transparent 70%);
        animation: ${load} 3s cubic-bezier(0.4, 0, 0.2, 1) infinite;
    }
`;

const Name = styled.div`
    background: #465c52;
    width: 4rem;
    height: 1rem;
    border-radius: 5px;
    position: relative;
    overflow: hidden;
    &::before {
        content: '';
        display: block;
        position: absolute;
        left: -150px;
        top: 0;
        height: 100%;
        width: 150px;
        background: linear-gradient(to right, transparent 30%, #009952 50%, transparent 70%);
        animation: ${load} 3s cubic-bezier(0.4, 0, 0.2, 1) infinite;
    }
`;

const UserSkeleton: React.FunctionComponent = () => {
    return (
        <Component>
            <Name />
            <Avatar />
        </Component>
    );
};

UserSkeleton.displayName = 'UserSkeleton';

export default UserSkeleton;
