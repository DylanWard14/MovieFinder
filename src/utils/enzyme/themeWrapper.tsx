import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
