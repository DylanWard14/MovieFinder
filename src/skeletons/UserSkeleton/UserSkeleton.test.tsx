import React from 'react';

import { renderWrapper } from '../../utils/enzyme/themeWrapper';
import UserSkeleton from './UserSkeleton';

describe('UserSkeleton', () => {
    describe('snapshot', () => {
        it('renders correctly', () => {
            const rendered = renderWrapper(<UserSkeleton />);

            expect(rendered).toMatchSnapshot();
        });
    });
});
