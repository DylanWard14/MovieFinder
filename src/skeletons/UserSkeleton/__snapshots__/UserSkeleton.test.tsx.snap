// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSkeleton snapshot renders correctly 1`] = `
<div
  class="UserSkeleton__Component-sc-173otu3-0 hbEMBZ"
>
  <div
    class="UserSkeleton__Name-sc-173otu3-2 dfxFNH"
  />
  <div
    class="UserSkeleton__Avatar-sc-173otu3-1 bRuhBC"
  />
</div>
`;
